core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -474,22 +519,171 @@ object ConsumerGroupCommand extends Logging {
+            case _ => null
+          }
+        }.toMap
+      } else if (opts.options.has(opts.resetShiftByOpt)) {
+        val currentCommittedOffsets = adminClient.listGroupOffsets(groupId)
+        partitionsToReset.map { topicPartition =>
+          val shiftBy = opts.options.valueOf(opts.resetShiftByOpt)
+          val currentOffset = currentCommittedOffsets.getOrElse(topicPartition,
+            throw new IllegalArgumentException(s"Cannot shift offset for partition $topicPartition since there is no current committed offset"))
+
+          val shiftedOffset = currentOffset + shiftBy
+          val newOffset = getLogEndOffset(topicPartition) match {
+            case LogOffsetResult.LogOffset(endOffset) if shiftedOffset > endOffset =>
+              warn(s"New offset ($shiftedOffset) is higher than latest offset. Value will be set to $endOffset")
+              endOffset
+
+            case _ => getLogStartOffset(topicPartition) match {
+              case LogOffsetResult.LogOffset(startOffset) if shiftedOffset < startOffset =>
+                warn(s"New offset ($shiftedOffset) is lower than earliest offset. Value will be set to $startOffset")
+                startOffset
+
+              case _ => shiftedOffset
+            }
+          }
+          (topicPartition, new OffsetAndMetadata(newOffset))
+        }.toMap
+      } else if (opts.options.has(opts.resetToDatetimeOpt)) {
+        partitionsToReset.map { topicPartition =>
+          val timestamp = getDateTime
+          val logTimestampOffset = getLogTimestampOffset(topicPartition, timestamp)
+          logTimestampOffset match {
+            case LogOffsetResult.LogOffset(offset) => (topicPartition, new OffsetAndMetadata(offset))
+            case _ => null
+          }
+        }.toMap
+      } else if (opts.options.has(opts.resetByDurationOpt)) {
+        partitionsToReset.map { topicPartition =>
+          val duration = opts.options.valueOf(opts.resetByDurationOpt)
+          val now = new Date()
+          val durationParsed = DatatypeFactory.newInstance().newDuration(duration)
+          durationParsed.negate().addTo(now)
+          val timestamp = now.getTime
+          val logTimestampOffset = getLogTimestampOffset(topicPartition, timestamp)
+          logTimestampOffset match {
+            case LogOffsetResult.LogOffset(offset) => (topicPartition, new OffsetAndMetadata(offset))
+            case _ => null
+          }
+        }.toMap
+      } else if (opts.options.has(opts.resetFromFileOpt)) {
+        val resetPlanPath = opts.options.valueOf(opts.resetFromFileOpt)
+        val resetPlanCsv = Utils.readFileAsString(resetPlanPath)
+        val resetPlan = parseResetPlan(resetPlanCsv)
+        partitionsToReset.map { topicPartition =>
+            if (resetPlan.keySet.contains(topicPartition))
+              (topicPartition, resetPlan(topicPartition))
+            else null
+        }.toMap
+      } else {
```

Comment:
Seems this case isn't actually allowed by the command. Maybe we can raise an `IllegalArgumentException` to make that explicit.